sale.py

```diff
@@ -441,6 +453,19 @@ def __setup__(cls):
         'invisible': Eval('type') != 'line',
     }, depends=['type'], required=True)
 
+    @fields.depends(
+        'product', 'unit', 'quantity', '_parent_sale.party',
+        '_parent_sale.currency', '_parent_sale.sale_date',
+        'delivery_mode', '_parent_sale.shop', '_parent_sale.shipment_address',
+        'warehouse', '_parent_sale.warehouse'
+    )
+    def on_change_delivery_mode(self):
+        """
+        This method can be overriden by downstream modules to compute taxes
```

Comment:
This method can be overridden by downstream modules to make changes according to delivery mode.
Like change taxes according to delivery mode.